server/routes/api/collections/collections.ts

```diff
@@ -60,29 +60,16 @@ router.post(
     const { user } = ctx.state.auth;
     authorize(user, "createCollection", user.team);
 
-    if (!index) {
-      const collections = await Collection.findAll({
-        where: {
-          teamId: user.teamId,
-          deletedAt: null,
-        },
-        attributes: ["id", "index", "updatedAt"],
-        limit: 1,
-        order: [
-          // using LC_COLLATE:"C" because we need byte order to drive the sorting
-          Sequelize.literal('"collection"."index" collate "C"'),
-          ["updatedAt", "DESC"],
-        ],
+    if (index) {
+      index = await removeIndexCollision(user.teamId, index, { transaction });
+    } else {
+      const first = await Collection.findFirstCollectionForUser(user, {
+        attributes: ["id", "index"],
         transaction,
       });
-
-      index = fractionalIndex(
-        null,
-        collections.length ? collections[0].index : null
-      );
+      index = fractionalIndex(null, first ? first.index : null);
     }
 
-    index = await removeIndexCollision(user.teamId, index);
```

Comment:
No point checking for index collision if we just calculated it

Reply:
Related: I'm wondering if `collectionIndexing` could be removed from `collections.list` endpoint 🤔 
```js
    const nullIndex = collections.findIndex(
      (collection) => collection.index === null
    );

    if (nullIndex !== -1) {
      const indexedCollections = await collectionIndexing(user.teamId);
      collections.forEach((collection) => {
        collection.index = indexedCollections[collection.id];
      });
    }
```

I don't see an obvious place where collection `index` is set to `null`(checked `collections.create` & `collections.update`). Perhaps, it's never set to `null` 🤔

I'm wondering if this indexing part was added as an [_in-place migration_ strategy](https://github.com/outline/outline/pull/2953/commits/e9789b95bd5595b5843752cbe1c2be4a9887caa2). In that case we can safely remove this if migration is achieved.

Reply:
> I'm wondering if this indexing part was added as an [in-place migration strategy](https://github.com/outline/outline/pull/2953/commits/e9789b95bd5595b5843752cbe1c2be4a9887caa2).

Yep, I think it doesn't do any harm to keep? We occasionally get workspaces that become active again after a year or two of no usage